apps/calendar/js/views/factory.js

```diff
@@ -0,0 +1,83 @@
+define(function(require, exports, module) {
+'use strict';
+
+var debug = require('debug')('viewFactory');
+var nextTick = require('next_tick');
+var snakeCase = require('snake_case');
+
+// FIXME: app is injected later, we can only remove this after Bug 1154988
+exports.app = null;
+exports._instances = Object.create(null);
+
+/**
+ * Initializes a view and stores
+ * a internal reference so when
+ * view is called a second
+ * time the same view is used.
+ *
+ * Makes an asynchronous call to
+ * load the script if we do not
+ * have the view cached.
+ *
+ *    // for example if you have
+ *    // a calendar view Foo
+ *
+ *    viewFactory.get('Foo', function(view) {
+ *      (view instanceof require('views/foo')) === true
+ *    });
+ *
+ * @param {String} name view name.
+ * @param {Function} view loaded callback.
+ */
+exports.get = function(name, cb) {
+  if (this._registered(name)) {
+    debug(`Found view named ${name}`);
+    this._get(name, cb);
+    return;
+  }
+
+  var path = 'views/' + snakeCase(name);
+
+  try {
+    // we try sync require first since it should be faster (skip an extra
+    // "tick"); our modules are bundled into few files during build time, so
+    // sync require should work in almost all the calls (will only fail if
+    // module was not defined/loaded before)
+    var Ctor = require(path);
```

Comment:
Also is there perhaps an rjs call that lets you know whether something is already loaded?